src/librustc/ty/print/pretty.rs

```diff
@@ -919,7 +919,15 @@ pub trait PrettyPrinter<'tcx>:
             },
             (ConstValue::Scalar(Scalar::Raw { data, .. }), ty::Char) =>
                 p!(write("{:?}", ::std::char::from_u32(data as u32).unwrap())),
-            (ConstValue::Scalar(_), ty::RawPtr(_)) => p!(write("{{pointer}}")),
+            (ConstValue::Scalar(value), ty::RawPtr(_)) => {
```

Comment:
```suggestion
            (ConstValue::Scalar(value), ty::RawPtr(_) | ty::Ref(..)) => {
```
That should take perfect care of references as well. ;)
(Well, assuming nested "or" patterns work already... not sure how far their implementation is along.)

Reply:
Looks like it's still feature gated. Your union idea worked with ZSTs, so I managed to add another test for refs and pushed another commit.

Reply:
You could also just add the feature gate in `librustc/lib.rs´. That crate already enables almost every feature gate under the sun anyway. ;)